libethereum/EthereumPeer.cpp

```diff
@@ -397,44 +397,51 @@ bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
 		host()->onPeerHeaders(dynamic_pointer_cast<EthereumPeer>(shared_from_this()), _r);
 		break;
 	}
-	case GetBlocksPacket:
+	case GetBlockBodiesPacket:
 	{
-		unsigned count = _r.itemCount();
-		clog(NetMessageSummary) << "GetBlocks (" << dec << count << "entries)";
+		unsigned count = static_cast<unsigned>(_r.itemCount());
+		clog(NetMessageSummary) << "GetBlockBodies (" << dec << count << "entries)";
 
 		if (!count)
 		{
-			clog(NetImpolite) << "Zero-entry GetBlocks: Not replying.";
+			clog(NetImpolite) << "Zero-entry GetBlockBodies: Not replying.";
 			addRating(-10);
 			break;
 		}
 		// return the requested blocks.
 		bytes rlp;
 		unsigned n = 0;
-		for (unsigned i = 0; i < min(count, c_maxBlocks) && rlp.size() < c_maxPayload; ++i)
+		auto numBodiesToSend = std::min(count, c_maxBlocks);
+		for (unsigned i = 0; i < numBodiesToSend && rlp.size() < c_maxPayload; ++i)
 		{
 			auto h = _r[i].toHash<h256>();
 			if (host()->chain().isKnown(h))
 			{
-				rlp += host()->chain().block(_r[i].toHash<h256>());
+				auto blockBytes = host()->chain().block(h);
+				RLP block{blockBytes};
+				auto transactions = block[1].data();
+				auto uncles = block[2].data();
+				auto inserter = std::back_inserter(rlp);
+				std::copy(transactions.begin(), transactions.end(), inserter);
+				std::copy(uncles.begin(), uncles.end(), inserter);
```

Comment:
I'm actually coping bytes from RLP of list of transactions, not transactions as RLP items. But I can see that a list containing 2 items is missing there.

What do you mean by "copy raw data" exactly?